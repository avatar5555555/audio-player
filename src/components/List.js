import React from 'react'
import PropTypes from 'prop-types'
import { ListGroup, ListGroupItem } from 'reactstrap'
import { FormattedMessage } from 'react-intl'
import styled from 'styled-components'

import { getTrackDuration } from 'src/services/time'

const MessageRoot = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100px;
  span {
    font-size: 25px;
    color: rgba(0, 0, 0, 0.6);
  }
`

const List = ({ items, onSelect }) => {
  const notFound = items.length === 0

  return notFound ? (
    <MessageRoot>
      <FormattedMessage id="notFound" />
    </MessageRoot>
  ) : (
    <ListGroup>
      {items.map(
        (
          { artistId, trackId, trackName, artistName, trackTimeMillis },
          index
        ) => {
          const isEven = index % 2 === 1
          const itemColor = isEven ? 'primary' : 'white'

          return (
            <ListGroupItem
              color={itemColor}
              key={`${artistId}-${trackId}`}
              tag="button"
              className="d-flex justify-content-between"
              onClick={() => onSelect(trackId)}
            >
              {`${index + 1}. ${artistName} - ${trackName}`}

              <div>{getTrackDuration(trackTimeMillis)}</div>
            </ListGroupItem>
          )
        }
      )}
    </ListGroup>
  )
}

List.propTypes = {
  items: PropTypes.array,
  onSelect: PropTypes.func.isRequired
}

List.defaultProps = {
  items: []
}

export default List
