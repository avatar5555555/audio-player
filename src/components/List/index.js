import React from 'react'
import PropTypes from 'prop-types'
import { ListGroup, ListGroupItem } from 'reactstrap'
import { FormattedMessage } from 'react-intl'
import styled from 'styled-components'

import { getTrackDuration } from 'src/services/time'

const MessageRoot = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100px;
  span {
    font-size: 25px;
    color: rgba(0, 0, 0, 0.6);
  }
`

const List = ({ items, onSelect, currentTrackId }) => {
  const notFound = items.length === 0

  return notFound ? (
    <MessageRoot>
      <FormattedMessage id="notFound" />
    </MessageRoot>
  ) : (
    <ListGroup>
      {items.map(
        (
          { artistId, trackId, trackName, artistName, trackTimeMillis },
          index
        ) => {
          const isCurrentTrack = currentTrackId === trackId
          const itemColor = isCurrentTrack ? 'primary' : 'white'

          return (
            <ListGroupItem
              color={itemColor}
              key={`${artistId}-${trackId}`}
              tag="button"
              className="d-flex justify-content-between"
              onClick={() => onSelect(trackId)}
            >
              {`${index + 1}. ${artistName} - ${trackName}`}

              <div>{getTrackDuration(trackTimeMillis)}</div>
            </ListGroupItem>
          )
        }
      )}
    </ListGroup>
  )
}

List.propTypes = {
  items: PropTypes.array,
  onSelect: PropTypes.func.isRequired,
  currentTrackId: PropTypes.string
}

List.defaultProps = {
  items: [],
  currentTrackId: ''
}

export default List
