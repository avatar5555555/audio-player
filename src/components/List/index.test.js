import React from 'react'
import { shallow, mount } from 'enzyme'
import toJSON from 'enzyme-to-json'

import List from './index'

const fakeProps = {
  onSelect: () => {},
  items: [{ trackId: '1' }]
}

describe('List', () => {
  it('renders', () => {
    shallow(<List {...fakeProps} />)
  })

  it('match snapshot', () => {
    expect(toJSON(shallow(<List {...fakeProps} />))).toMatchSnapshot()
  })

  it('color primary when currentTrack is the same as trackId', () => {
    const wrap = () => mount(<List {...fakeProps} currentTrackId={'1'} />)

    expect(
      wrap()
        .find('ListGroupItem')
        .props()
    ).toHaveProperty('color', 'primary')
  })

  it('color white when currentTrack is not the same as trackId', () => {
    const wrap = () => mount(<List {...fakeProps} currentTrackId={'2'} />)

    expect(
      wrap()
        .find('ListGroupItem')
        .props()
    ).toHaveProperty('color', 'white')
  })
})
