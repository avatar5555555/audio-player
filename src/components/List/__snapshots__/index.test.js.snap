// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List match snapshot 1`] = `
<ListGroup
  tag="ul"
>
  <ListGroupItem
    className="d-flex justify-content-between"
    color="white"
    key="undefined-1"
    onClick={[Function]}
    tag="button"
  >
    1. undefined - undefined
    <div>
      0:00
    </div>
  </ListGroupItem>
</ListGroup>
`;
